app/javascript/lib/layout-utils.js

```diff
@@ -0,0 +1,154 @@
+
+  // Ensure tooltip doesn't get cut off from above
+  studyHeader.setAttribute('data-placement', 'bottom')
+}
+
+/** Remove tooltip from header for study title, e.g. when it *not* truncated */
+function removeStudyTitleTooltip(studyHeader) {
+  if (!studyHeader.hasAttribute('data-toggle')) {return}
+  studyHeader.removeAttribute('data-toggle')
+  studyHeader.removeAttribute('data-original-title')
+  studyHeader.removeAttribute('data-placement')
+}
+
+/** Slightly decrease size of font and left padding */
+function shrinkStudyTitle(studyHeader) {
+  // Decrease font size by 1px and padding at left by 1/2 original
+  const fontSize = pxToNumber(getStyle(studyHeader, 'font-size')) // e.g. '14px' -> 14
+  const smallerFontSize = `${fontSize - 1}px`
+  studyHeader.style.fontSize = smallerFontSize
+  const paddingLeft = pxToNumber(getStyle(studyHeader, 'padding-left'))
+  const smallerPaddingLeft = `${paddingLeft / 2}px`
+  studyHeader.style.paddingLeft = smallerPaddingLeft
+}
+
+/** Undo `shrinkStudyTitle` */
+function unshrinkStudyTitle(studyHeader) {
+  studyHeader.removeAttribute('style')
+}
+
+/** Determine if study title has x-overflow in study header container */
+function getIsTitleTruncated(studyHeader) {
+  const studyHeaderWidth = studyHeader.clientWidth
+  const titleText = studyHeader.innerText
+  const titleFont = getStyle(studyHeader, 'font')
+  const studyTitleWidth = getTextSize(titleText, titleFont).width
+
+  const isTitleTruncated = studyTitleWidth > studyHeaderWidth
+
+  return isTitleTruncated
+}
+
+/**
+ * Mitigate truncation edge case for very long titles
+ *
+ * If study title is smaller than its container, and user's screen is basically maximized,
+ * then slightly decrease font size of study title, and slightly enlarge container.
+ */
+export function mitigateStudyOverviewTitleTruncation() {
+  if (window.SCP.analyticsPageName !== 'site-study') {return}
+
+  const studyHeader = document.querySelector('.study-header')
+  const titleText = studyHeader.innerText
+  const isTruncated = getIsTitleTruncated(studyHeader)
+  if (isTruncated) {
+    if (!studyHeader.hasAttribute('style')) {shrinkStudyTitle(studyHeader)}
+    const isStillTruncated = getIsTitleTruncated(studyHeader)
+    if (isStillTruncated) {
+      addStudyTitleTooltip(studyHeader, titleText)
+      studyHeader.classList.add('right-border-0-5')
+    }
+  } else {
+    if (studyHeader.hasAttribute('style')) {
+      unshrinkStudyTitle(studyHeader)
+      const isNowTruncated = getIsTitleTruncated(studyHeader)
+      if (isNowTruncated) {
+        shrinkStudyTitle(studyHeader)
+      }
+    }
+    removeStudyTitleTooltip(studyHeader)
+    studyHeader.classList.remove('right-border-0-5')
+  }
+
+  const jqStudyHeader = $('.study-header')
+  if (jqStudyHeader.tooltip) {jqStudyHeader.tooltip('hide')}
+}
+
+/**
+ * Set global header end width, and mitigate long study titles on narrow screens
+ */
+export function adjustGlobalHeader() {
```

Comment:
FWIW I checked on production, and the longest study title we have is `Intra-tumoral injection of caerin 1.1 and 1.9 peptides in vaccinated TC-1 tumour bearing mice with PD-1 blockade modulates macrophage heterogeneity and the activation of CD8+ T cells in the tumour microenvironment`.  I renamed one of my local studies to that and this dealt with that just fine.